### PR TITLE
easyeffects: 6.1.3 -> 6.2.3

### LINKED ISSUE #155588: Update easyeffects to 6.2.1
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: easyeffects

current version: 6.1.3
desired version: 6.2.1

###### Notify maintainers

maintainers: @jtojnar 

###### Note for maintainers

Please tag this issue in your PR.

### LINKED ISSUE #158476: Pipewire Update Broke EasyEffects
### Describe the bug
After the recent update to Pipewire ( two commits on master branch: 3beb2a3aff32a380ca9b511270e4044ae283bf08 40af850b99185933bca50ad533bbfbdbf4118c1b ) EasyEffects always crashes when any sound playback is attempted.

### Steps To Reproduce
1. Use Pipewire
2. Install EasyEffects and maybe add an effect ( I did not think to test an empty chain ).

### Expected behavior
Playback of appropriately filtered sound.

### Additional context
I use bs2b and equalizer. After I reverted the two commits in my fork of nixpkgs, the program works properly.

### Notify maintainers
@jansol @jtojnar @Kranzes 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/nlmr383l7l3lbhmy6gm5rc7z1ldy2l3n-nix-info
copying path '/nix/store/nlmr383l7l3lbhmy6gm5rc7z1ldy2l3n-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.16.7, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.7.0pre20220127_558c4ee`
 - channels(root): `"nixos-22.05pre347947.689b76bcf36"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

### PR DESCRIPTION
**Pull request by MarekPasnikowski**
1. Updated to the newest version.
2. Added some build and runtime dependencies according to build errors.

###### Motivation for this change
Fixes: #158476
Fixes: https://github.com/NixOS/nixpkgs/issues/155588

https://github.com/wwmm/easyeffects/blob/v6.2.3/CHANGELOG.md

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jtojnar** [comment]
The `substituteInPlace` can be removed.

File: pkgs/applications/audio/easyeffects/default.nix [hunk 1]
```diff
@@ -49,15 +52,18 @@ stdenv.mkDerivation rec {
     ninja
     pkg-config
     python3
+    tbb
```
Comment (jtojnar) [inline pkgs/applications/audio/easyeffects/default.nix:55]: How did you determine it goes to `nativeBuildInputs`?

File: pkgs/applications/audio/easyeffects/default.nix [hunk 2]
```diff
@@ -49,15 +52,18 @@ stdenv.mkDerivation rec {
...
     wrapGAppsHook4
   ];
 
   buildInputs = [
     fftwFloat
+    fmt
     glib
     glibmm
     gtk4
     gtkmm4
```
Comment (jtojnar) [inline pkgs/applications/audio/easyeffects/default.nix:65]: From [changelog](https://github.com/wwmm/easyeffects/blob/v6.2.3/CHANGELOG.md):

> GTKMM and GLIBMM are not a dependency anymore. We now use gtk4 directly.

File: pkgs/applications/audio/easyeffects/default.nix (hunk 1 above)
Comment (Obikawa) [reply pkgs/applications/audio/easyeffects/default.nix:55]: Good catch - when I made this change, I was under impression that native inputs meant "build time" dependencies. But after thinking about it in the cross-compilation context (which I had learned later) I am not sure what should go where anymore. Is there an algorithm to accurately decide this?

Comment (jtojnar) [reply pkgs/applications/audio/easyeffects/default.nix:54]: That is about right `nativeBuildInputs` is about dependencies used, or rather, executed at build time. So mostly programs used for locating dependencies, build tools, linkers, documentation generators… `tbb` is a library and while it is possible to build and run an executable linking a library at build time, it is quite rare. So unless you have a reason to suspect a library to be used this way (e.g. `libxslt` contains `xsltproc` program often used for building man pages from DocBook files), it is most likely a runtime dependency belonging to `buildInputs`.

See also https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md

File: pkgs/applications/audio/easyeffects/default.nix [hunk 3]
```diff
@@ -4,11 +4,12 @@
 , fetchFromGitHub
 , calf
 , fftwFloat
+, fmt
 , glib
 , glibmm
```
Comment (jtojnar) [inline pkgs/applications/audio/easyeffects/default.nix:9]: This one too.

Comment (jtojnar) [reply pkgs/applications/audio/easyeffects/default.nix:9]: > anonymous function at /home/jtojnar/Projects/nixpkgs/pkgs/applications/audio/easyeffects/default.nix:1:1 called with unexpected argument 'glibmm'

means that you also need to remove `glibmm` from being passed in `all-packages.nix`.

File: pkgs/applications/audio/easyeffects/default.nix (hunk 1 above)
Comment (Obikawa) [reply pkgs/applications/audio/easyeffects/default.nix]: Is it safe to always put any new dependency in buildInputs first? And only when it breaks move it to nativeBuildInputs?

File: pkgs/applications/audio/easyeffects/default.nix [hunk 4]
```diff
@@ -69,14 +70,15 @@ stdenv.mkDerivation rec {
...
+    tbb
     zita-convolver
   ];
 
   postPatch = ''
     chmod +x meson_post_install.py
     patchShebangs meson_post_install.py
     # https://github.com/wwmm/easyeffects/pull/1205
-    substituteInPlace meson_post_install.py --replace "gtk-update-icon-cache" "gtk4-update-icon-cache"
+    # substituteInPlace meson_post_install.py --replace "gtk-update-icon-cache" "gtk4-update-icon-cache"
```
Comment (jtojnar) [inline pkgs/applications/audio/easyeffects/default.nix:81]: You can just drop this. This change was merged in 6.1.4.

File: pkgs/applications/audio/easyeffects/default.nix (hunk 1 above)
Comment (Obikawa) [reply pkgs/applications/audio/easyeffects/default.nix]: > so it should not be left in the code

Does `it` reference to the `strictDeps` call? I am roughly understanding what you wrote as "Use `strictDeps` in testing, but make sure it is removed in the final code."

File: pkgs/applications/audio/easyeffects/default.nix (hunk 4 above)
Comment (Obikawa) [reply pkgs/applications/audio/easyeffects/default.nix]: Dropped.

**Comment by jtojnar** [comment]
Now it fails on missing

```
src/meson.build:122:0: ERROR: Dependency "sigc++-3.0" not found, tried pkgconfig
```

**Comment by jtojnar** [comment]
Could you please also squash everything into `easyeffects: 6.1.3 -> 6.2.3`?

**Comment by Obikawa** [comment]
Understood. I have to learn how to squash in GitHub and add libsigcxx30.

I need to do something time-sensitive in fifteen minutes, so expect a longer response time while I get everything in order.

**Comment by Obikawa** [comment]
With regards to squashing:

After reading some documentation and internet articles, I got the impression that this option is available only to those with write access to a given repository. I do not have this capability on this repository.

Do you not have the `squash and merge` option available?

**Comment by jtojnar** [comment]
You only need to squash in your branch in your repository (https://github.com/Obikawa/nixpkgs/tree/easyeffects-update). You can use `git rebase --interactive master` to rebase onto `master` branch (or whatever other branch you want).

In this case, using “Rebase and squash” will work since we want to produce only a single logical commit, it is useless for anything else. The disadvantage is that you will not be able to use `git branch --merged` to find and delete merged branches in your local repository since “rebase and squash” will not update your own repository. But if you do not use that workflow, I guess that is fine.

**Comment by Obikawa** [comment]
It is fine:

1. The branch is only for the purpose of getting the update merged with upstream and will be thrown away after that.
2. I have had trouble using my hardware key (Trezor) to authenticate with GitHub, so I settled with web-only workflow. The GitHub's web interface does not seem to have any option to squash a branch.

**Comment by jtojnar** [comment]
Thanks.

**Comment by jtojnar** [comment]
Since pipewire is backported we need to backport this as well.
